stream-access-control/access-control-core/src/main/java/com/backbase/stream/service/AccessGroupService.java

```diff
@@ -1228,11 +1225,23 @@ private Mono<JobRole> createJobRole(StreamTask streamTask, ServiceAgreement serv
         presentationIngestFunctionGroup.setExternalServiceAgreementId(serviceAgreement.getExternalId());
         presentationIngestFunctionGroup.setMetadata(jobRole.getMetadata());
 
-        //since ReferenceJobRole class was removed, now all job roles have the same class, and
-        //reference job role can be created only for MSA, for CSA it failed
-        if(BooleanUtils.isTrue(serviceAgreement.getIsMaster())) {
-            log.debug("Creating a Reference Job Role.");
-            presentationIngestFunctionGroup.setType(PresentationIngestFunctionGroup.TypeEnum.TEMPLATE);
+        //Logic to map the job roles types
+        final var jobRoleType = jobRole.getFunctionGroups().getFirst().getType();
```

Comment:
a have concerns with null pointer exceptions here

Reply:
Since Job role is a set of functions, so having a job role without any function group is anyway wrong. We could of course fail with a better exception than NPE if that makes better sense

Reply:
Updated